website/docs/extensions/write/index.md

```diff
@@ -0,0 +1,135 @@
+---
+sidebar_position: 1
+title: Writing
+description: Writing my first extension
+tags: [podman-desktop, extension, writing]
+keywords: [podman desktop, extension, writing]
+---
+
+# Writing your first extension
+
+Podman Desktop extensions rely on `JavaScript` or `TypeScript`.
+
+Extensions are self-contained, all runtime dependencies are inside the final binary.
+
+#### Scaffold the project
+
+1. Setup a Node.js project with `TypeScript`. Use for example `npm init` command.
+
+1. Bring the development dependency to Podman Desktop API. It allows to import the API and register the contributions.
+
+  Note that it is a dependency that brings only the definition of the API. It will not fetch any Podman Desktop internals.
+
+  ```json
+    "devDependencies": {
+      "@podman-desktop/api": "latest",
+    }
+  ```
+
+1. Setup the required metadata for the extension
+
+  ```json
+  {
+    "name": "my-extension",
+    "displayName": "My Hello World extension",
+    "description": "How to write my first extension",
+    "version": "0.0.1",
+    "icon": "icon.png",
+    "publisher": "benoitf",
+    "engines": {
+      "podman-desktop": "latest"
+    }
+  }
+  ```
+
+  In the `engines` section, describe the version of Podman Desktop that runs this extension.
+
+1. Include the main entry point of the extension
+
+  ```json
+    "main": "./dist/extension.js",
+  ```
+
+1. Add Hello World command contribution
+
+  ```json
+    "contributes": {
+      "commands": [
+        {
+          "command": "my.first.command",
+          "title": "My First Extension: Hello World"
+        }
+      ]
+    },
+  ```
+
```

Comment:
The `icon.png` file is mentioned above, and in the publication procedure. Is it the right place to mention to add it to the project?

```suggestion

1. Add an `icon.png` file to the project.
```